rfc8446.xml

```diff
@@ -4221,18 +4721,37 @@ rejecting 0-RTT but MUST NOT abort the handshake.</t>
 of the ClientHello. If the ClientHello contains multiple
 PSK identities, then an attacker can create multiple ClientHellos
 with different binder values for the less-preferred identity on the
-assumption that the server will not verify it, as recommended
-by <xref target="pre-shared-key-extension"/>.
-I.e., if the
+assumption that the server will not verify it as recommended
+by <xref target="pre-shared-key-extension"/>. That is, if the
 client sends PSKs A and B but the server prefers A, then the
 attacker can change the binder for B without affecting the binder
 for A. If the binder for B is part of the storage key,
 then this ClientHello will not appear as a duplicate,
 which will cause the ClientHello to be accepted, and may
 cause side effects such as replay cache pollution, although any
 0-RTT data will not be decryptable because it will use different
-keys. If the validated binder or the ClientHello.random
-are used as the storage key, then this attack is not possible.</t>
+keys.
+
+<!-- [rfced] Section 8.2:  We had trouble following this sentence.
+If the suggested text is not correct, please clarify.
+
+Original:
+ If the binder for B is part of
+ the storage key, then this ClientHello will not appear as a
+ duplicate, which will cause the ClientHello to be accepted, and may
+ cause side effects such as replay cache pollution, although any 0-RTT
+ data will not be decryptable because it will use different keys.
+
+Suggested:
+ If the binder for B is part of
+ the storage key, then this ClientHello will not appear as a
+ duplicate, causing the ClientHello to be accepted and possibly
+ also causing side effects such as replay cache pollution, although
+ any 0-RTT data will not be decryptable because it will use different
+ keys. -->
```

Comment:
We couldn't tell what causes the ClientHello to be accepted (maybe because the binder for B is part of the storage key, or maybe because the ClientHello won't appear as a duplicate).  If the text will be clear to other readers, we're happy to withdraw this question.

Reply:
The ClientHello would be accepted because the attempt to look for a previous copy in the cache would fail (since the thing being used as the lookup key has been modified by the attacker), and the point of this operation is to prevent duplicates.